.github/workflows/remove-wheels.yml

```diff
@@ -0,0 +1,75 @@
+name: Remove old wheels
+
+on:
+  # Run daily at 1:23 UTC
+  schedule:
+  - cron:  '23 1 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Needed for micromamba pickup
+defaults:
+  run:
+    shell: bash -l {0}
+
+env:
+  N_LATEST_UPLOADS: 5
+  ANACONDA_USER: "scientific-python-nightly-wheels"
+
+jobs:
+  remove:
+
+    runs-on: ubuntu-latest
+    environment:
+      name: remove-old-wheels
```

Comment:
Using a GHA environment to further restrict the possibility of accessing any API tokens.

Someone who has org admin permissions needs to create a GitHub Actions environment (https://github.com/scientific-python/upload-nightly-action/settings/environments) named "remove-old-wheels" and in there create an environment secret named `ANACONDA_TOKEN`. The value of `ANACONDA_TOKEN` can be set to the value of the `github-actions` Anaconda Cloud API token fund under https://anaconda.org/scientific-python-nightly-wheels/settings/access.

Reply:
Yup, a comment is a good idea here. I'll add one.

Reply:
@matthewfeickert @stefanv @tupui I marked it as a draft to see if folks could take one last look (I don't want to accidentally want to delete stuff).

I followed the instructions below and I believe this ready to merge.

> Using a GHA environment to further restrict the possibility of accessing any API tokens.
> 
> Someone who has org admin permissions needs to create a GitHub Actions environment (https://github.com/scientific-python/upload-nightly-action/settings/environments) named "remove-old-wheels" and in there create an environment secret named `ANACONDA_TOKEN`. The value of `ANACONDA_TOKEN` can be set to the value of the `github-actions` Anaconda Cloud API token fund under https://anaconda.org/scientific-python-nightly-wheels/settings/access.

Reply:
> I followed the instructions below and I believe this ready to merge.

Great! Yeah, if the https://github.com/scientific-python/upload-nightly-action/settings/environments view looks similar to the following (from my fork) then this should be good to go.

![env-view](https://github.com/scientific-python/upload-nightly-action/assets/5142394/bface045-acc8-4024-a182-c79b7bafc354)

Reply:
It does!!

Reply:
![2023-06-08T10:04:24,228096276-07:00](https://github.com/scientific-python/upload-nightly-action/assets/123428/40daf283-ee31-4fe7-b98a-3721387d5ce9)

Reply:
I am +1 here. Go for it I would say 😃